packages/cf.js/src/client.ts

```diff
@@ -0,0 +1,87 @@
+import * as uuid from "uuid";
+
+import { AppFactory } from "./app-factory";
+import { AppInstance } from "./app-instance";
+import {
+  AppDefinition,
+  MessageDataQuery,
+  NodeMessage,
+  NodeMessageType,
+  NodeProvider,
+  QueryType
+} from "./structs";
+
+export enum ClientEventType {
+  INSTALL = "install",
+  PROPOSE_INSTALL = "proposeInstall",
+  REJECT_INSTALL = "rejectInstall"
+}
```

Comment:
Are these the message types the NodeProvider expects? Can we prefix them with `cf_`?

Reply:
@snario Curious to know if we're prefixing them for any reason. Is name collision possible in this context?

Reply:
Mirroring web3.